src/org/skypro/skyshop/basket/ProductBasket.java

```diff
@@ -2,23 +2,23 @@
 
 import org.skypro.skyshop.product.Product;
 
-import java.util.Iterator;
-import java.util.LinkedList;
-import java.util.List;
+import java.util.*;
 
 public class ProductBasket {
-    private List<Product> products = new LinkedList<>();
+    private Map<String, ArrayList<Product>> products = new HashMap<>();
```

Comment:
```suggestion
    private Map<String, List<Product>> products = new HashMap<>();
```
не забываем про полиморфизм